main/src/com/google/refine/expr/functions/strings/ParseUri.java

```diff
@@ -0,0 +1,89 @@
+package com.google.refine.expr.functions.strings;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.refine.expr.EvalError;
+import com.google.refine.grel.Function;
+import com.google.refine.util.ParsingUtilities;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Parse a URI string into its components and returns a JSON object with the following keys:
+ * scheme, authority, path, query, fragment, host, port.
+ */
+public class ParseUri implements Function {
+
+    @Override
+    public Object call(Properties bindings, Object[] args) {
+        if(args.length == 1 && args[0] instanceof String) {
+            String s = (String)args[0];
+            try {
+                URL url = new URL(s);
+                URI uri = url.toURI();
+                ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
I think Jackson's docs recommend not to instantiate an `ObjectMapper` too often and instead rely on a statically stored one. We have one you can use at `ParsingUtilities.mapper` (you are actually already using that below, so you might just be able to delete this instance :) )

Reply:
Oh! Yes, thanks 😁